depthai_examples/ros1_src/stereo_inertial_publisher.cpp

```diff
@@ -374,32 +376,57 @@ int main(int argc, char** argv) {
                                                         nnPath);
 
     std::shared_ptr<dai::Device> device;
-    std::vector<dai::DeviceInfo> availableDevices = dai::Device::getAllAvailableDevices();
-
-    std::cout << "Listing available devices..." << std::endl;
-    for(auto deviceInfo : availableDevices) {
-        std::cout << "Device Mx ID: " << deviceInfo.getMxId() << std::endl;
-        if(deviceInfo.getMxId() == mxId) {
-            if(deviceInfo.state == X_LINK_UNBOOTED || deviceInfo.state == X_LINK_BOOTLOADER) {
-                isDeviceFound = true;
-                if(poeMode) {
-                    device = std::make_shared<dai::Device>(pipeline, deviceInfo);
-                } else {
-                    device = std::make_shared<dai::Device>(pipeline, deviceInfo, usb2Mode);
-                }
-                break;
-            } else if(deviceInfo.state == X_LINK_BOOTED) {
-                throw std::runtime_error("ros::NodeHandle() from Node \"" + pnh.getNamespace() + "\" DepthAI Device with MxId  \"" + mxId
-                                         + "\" is already booted on different process.  \"");
+    if(useWithIP) // Connecting to a camera with specific IP, requires static IP config beforehand
+    {
+        auto deviceInfo = dai::DeviceInfo(ipAddress);
+        if(deviceInfo.state == X_LINK_ANY_STATE || deviceInfo.state == X_LINK_UNBOOTED || deviceInfo.state == X_LINK_BOOTLOADER || deviceInfo.state == X_LINK_FLASH_BOOTED) {
```

Comment:
PoE device can be launched only if it is in Bootloader Mode I think. IIRC.

Reply:
@saching13 I tested this with POE device too and sometimes it goes to X_LINK_ANY_STATE. Also, if you update the latest bootloader, it takes FLASH_BOOTED, so to be on the safe side, I included all the conditions, but if you think some of them is not necessary, then I will remove unnecessary ones.

Reply:
is this the case ? @themarpe

Reply:
XLink any state would be an issue querying the information from the device. We should fix that in core. Fo you have a repro @samialperen ?

Flash booted state happens when a pipeline is flashed onto device to run standalone on boot. That is a valid case to then stop that pipeline and start a new one

Reply:
@themarpe @saching13 I have OAK-D version 1, OAK-D-PRO-W POE, OAK-D-PRO USB, OAK-D-PRO-W USB. If you can share more context about what is expected from each device (or device type USB vs poe), I can test this one and share the results here.

Reply:
If queried you should see unbooted, bootloader, and booted if already running. 
Check protocol for type of connection (usb/tcpip)